read-fonts/src/tables/variations.rs

```diff
@@ -448,36 +452,53 @@ impl DeltaRunType {
 /// Implements the logic for iterating over the individual runs
 #[derive(Clone, Debug)]
 pub struct DeltaRunIter<'a> {
-    remaining: u8,
+    limit: Option<usize>, // when None, consume all available data
+    remaining_in_run: u8,
     value_type: DeltaRunType,
     cursor: Cursor<'a>,
 }
 
 impl<'a> DeltaRunIter<'a> {
-    fn new(cursor: Cursor<'a>) -> Self {
+    fn new(cursor: Cursor<'a>, limit: Option<usize>) -> Self {
         DeltaRunIter {
-            remaining: 0,
+            limit,
+            remaining_in_run: 0,
             value_type: DeltaRunType::I8,
             cursor,
         }
     }
+
+    pub(crate) fn end(mut self) -> Cursor<'a> {
+        while self.next().is_some() {}
+        self.cursor
+    }
 }
 
 impl Iterator for DeltaRunIter<'_> {
     type Item = i32;
 
     fn next(&mut self) -> Option<Self::Item> {
+        if let Some(limit) = self.limit {
+            if limit == 0 {
+                debug_assert!(
```

Comment:
a `debug_assert!`???

Reply:
I miss having log, I want the equivalent of warn. Maybe debug_assert is a little more firepower than warn ... removed it .